src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -1070,6 +1070,9 @@ public virtual void IntersectWith(IEnumerable<T> other)
             if (Count == 0)
                 return;
 
+            if (other == this)
+                return;
```

Comment:
Since you've removed the DoRemove that you'd added that was causing the exception, this `if (other == this) return;` addition probably isn't necessary anymore, but it also doesn't really hurt anything.